python/lsst/ts/externalscripts/maintel/focus_telescope.py

```diff
@@ -0,0 +1,396 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["FocusTelescope"]
+
+import abc
+import asyncio
+import types
+import typing
+import warnings
+from collections import defaultdict
+
+import numpy as np
+import yaml
+from astropy.table import QTable
+from lsst.daf.butler import Butler
+from lsst.ts import salobj
+from lsst.ts.observatory.control import BaseCamera
+from lsst.ts.observatory.control.maintel.lsstcam import LSSTCam
+from lsst.ts.observatory.control.maintel.mtcs import MTCS
+
+try:
+    from lsst.ts.wep.task.fitDonutRadiusTask import (
+        FitDonutRadiusTask,
+        FitDonutRadiusTaskConfig,
+    )
+
+except ImportError:
+    warnings.warn("Cannot import required libraries. Script will not work.")
+
+CMD_TIMEOUT = 100
+
+
+class FocusTelescope(salobj.BaseScript, metaclass=abc.ABCMeta):
+    """Focus telescope"""
+
+    def __init__(self, index=1, descr="") -> None:
+        super().__init__(
+            index=index,
+            descr=descr,
+        )
+
+        self.mtcs = None
+        self._camera = None
+
+        self.butler = None
+
+    @property
+    def camera(self) -> BaseCamera:
+        if self._camera is not None:
+            return self._camera
+        else:
+            raise RuntimeError("Camera not defined.")
+
+    @camera.setter
+    def camera(self, value: BaseCamera | None) -> None:
+        self._camera = value
+
+    async def configure_butler(self) -> None:
+        """Configure the butler."""
+        if self.butler is None:
+            self.butler = Butler("/repo/LSSTCam")
+        else:
+            self.log.debug("Butler already defined, skipping.")
+
+    async def configure_camera(self) -> None:
+        """Handle creating Camera object and waiting for remote to start."""
+        if self._camera is None:
+            self.log.debug("Creating Camera.")
+
+            self._camera = LSSTCam(
+                self.domain,
+                log=self.log,
+                tcs_ready_to_take_data=self.mtcs.ready_to_take_data,
+            )
+            await self._camera.start_task
+        else:
+            self.log.debug("Camera already defined, skipping.")
+
+    async def configure_tcs(self) -> None:
+        """Handle creating MTCS object and waiting for remote to start."""
+        if self.mtcs is None:
+            self.log.debug("Creating MTCS.")
+            self.mtcs = MTCS(
+                domain=self.domain,
+                log=self.log,
```

Comment:
You should probably limit the resources here... 

import the `MTCSUsages`:

```
from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
```

in line 38 than here do:

```
            self.mtcs = MTCS(
                domain=self.domain,
                log=self.log,
                intended_usage=MTCSUsages.Slew | MTCSUSages.StateTransition,
            )
```